src/main/java/uk/gov/cabinetoffice/csl/service/user/UserProfileService.java

```diff
@@ -75,8 +80,28 @@ public void setProfession(String uid, Long professionId) {
         }
     }
 
+    public void setOrganisationalUnit(String uid, Long organisationalUnitId) {
+        Optional<OrganisationalUnit> optOrganisationalUnit = organisationalUnitListService
+                .getAllOrganisationalUnits()
+                .getOrganisationalUnits()
+                .stream()
+                .filter(g -> g.getId().equals(organisationalUnitId))
```

Comment:
Could add a new method to the `organisationalUnitListService` to fetch by ID for this

Reply:
This code is redundant.